### PR TITLE
Add issue/PR template, DEVELOPING.md

### PR DESCRIPTION
**Pull request by brianlheim**
## Purpose/Motivation

This PR adds an issue template, PR template, and the DEVELOPING.md file
we've had in incubation on the Wiki for awhile. I want to make some more
changes after this to tidy up our front-facing documentation; this is a
start.

## Types of Changes

- Documentation (non-code change which corrects or adds documentation for existing features)

## Checklist

- [x] Updated documentation, if necessary

- [x] This PR is ready for review

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+  - UGens should be fully documented, with a clear explanation of what it does and an appropriate collection of
+    examples. Be sure to document which UGen inputs are modulatable at which rates.
+  - UGens should be efficient. SuperCollider takes pride in being easy on the CPU, and UGens should help support that
+    reputation.
+  - The UGen should be deemed useful enough to the general SC user base.
+  - Each UGen should be in a separate C++ source file (or multiple source files if it's really that long).
+  - Don't leave any unnecessary print statements lying around.
+- **Safety:**
+  - Check input rates in the sclang class.
+  - UGens should be real-time safe. Ross Bencina's ["Real-time audio programming 101"]
```
Comment (patrickdupuis) [inline DEVELOPING.md]: This link is broken!?! Ross Bencina, say it ain't so!

File: DEVELOPING.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    (http://www.rossbencina.com/code/real-time-audio-programming-101-time-waits-for-nothing) is required reading.
+    Here is a quick and incomplete blacklist of dangerous features:
+    - `malloc`/`free`/`new`: use `RTAlloc` instead.
+    - `throw`/`catch`: use return codes instead.
+    - `dynamic_cast`
+    - System calls. These are complicated to handle and require use of an NRT thread.
+  - Any calls to `RTAlloc` should be protected from `RTAlloc` returning a null pointer. This usually happens when there
```
Comment (patrickdupuis) [inline DEVELOPING.md]: Better written:
`Any calls to `RTAlloc` should be protected from returning a null pointer.`

File: DEVELOPING.md [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    isn't enough real-time memory left, and results in a **server crash** if the code is not protected from attempts to
```
Comment (patrickdupuis) [inline DEVELOPING.md]: "...isn't enough real-time memory left, _resulting_ in a **server crash** if the code is not protected from attempts to ... "

File: DEVELOPING.md [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+    read from an unallocated buffer.
+  - The Ctor sample should be initialized. If this is not done, very nasty bugs can occur.
+  - Zap dangerous values (subnormals, infinities, nans) in feedback loops to 0. SC provides a `zapgremlins` function
```
Comment (patrickdupuis) [inline DEVELOPING.md]: Reads better if you write it as `zero`.

File: DEVELOPING.md [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    that does this for you.
+- **Utility:**
+  - UGens should have both `.ar` and `.kr` methods if applicable.
+  - Sample rate and block size independence should be maintained if applicable. For example, audio UGens shouldn't sound
+    radically different if the sample rate is increased.
+  - For audio UGens, control-rate inputs should be interpolated if applicable.
+  - Don't arbitrarily make certain inputs nonmodulatable just for programming convenience -- carefully anticipate what's
+    worth modulating. Either way, don't forget to document it.
+- **Modernization and deprecated features:**
+  - When writing a new UGen from scratch, it is recommended to use the modern C++ style seen in `SC_PlugIn.hpp`.
```
Comment (patrickdupuis) [inline DEVELOPING.md]: Maybe add a link to `SC_PlugIn.hpp`?

File: DEVELOPING.md [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+* http://supercollider.s3.amazonaws.com/builds/supercollider/supercollider/win32/SC-Windows-x86-8c3563a8065cb623087f267dfe50e228224a4572.zip
+* http://supercollider.s3.amazonaws.com/builds/supercollider/supercollider/win64/SC-Windows-x64-8c3563a8065cb623087f267dfe50e228224a4572.zip
+
+A build for a specific commit may not always be available: for instance, if the build was cancelled early or failed to
+complete.
+
+Boost Update Script
+-------------------
+
+Location: `external_libraries/extract_boost.sh`, `external_libraries/boost_sc_changes.patch`
```
Comment (patrickdupuis) [inline DEVELOPING.md]: Link to files?

File: DEVELOPING.md [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+
+### Purpose
+
+Update Boost libraries packaged with SuperCollider, and apply the SuperCollider organization's patches for Boost.
+
+### Usage
+
+Should be run as soon as possible after a new release of Boost. See instructions in `external_libraries/README_BOOST.md`
```
Comment (patrickdupuis) [inline DEVELOPING.md]: Link to file?

File: DEVELOPING.md [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+
+qpm is a Python-based Quarks package manager and test runner for SuperCollider. It is used by our CI services to run the
+SuperCollider-based test suite. See https://github.com/scztt/qpm/tree/qpm-unit for more information.
+
+Changelog-to-schelp script
+--------------------------
+
+Location: https://gist.github.com/brianlheim/443ae188dee8f7a85e7f34c04cc66d2b
+
+### Purpose
```
Comment (patrickdupuis) [inline DEVELOPING.md]: We don't really need this header.

File: DEVELOPING.md [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+	The integer index of the column to modify
+ARGUMENT:: width
+	Integer width in pixels
+```
+
+Project Structure
+=================
+
+- **`.github`**: PR and issue template for GitHub
+- **`.travis`**: scripts for Travis CI
```
Comment (patrickdupuis) [inline DEVELOPING.md]: Shouldn't these two files be listed in the "important files..." section below?

**Comment by snappizz** [comment]
@patrickdupuis these are the kinds of changes where it's easiest if you push directly to brian's branch, or wait for it to be merged and make the changes in a second PR

**Comment by patrickdupuis** [comment]
OK, I'll see if I can do the important ones tonight.

**Comment by brianlheim** [comment]
Made some changes as requested by Patrick. Personally I do not really like the idea of inserting links; it obscures readability when reading as plaintext, but you can add it in a separate PR if you want, OK?

**Comment by patrickdupuis** [comment]
All good. Sorry I couldn't tell the difference between a folder and a file :/

**Pull request merged**